libs/queue/src/main.sw

```diff
@@ -0,0 +1,98 @@
+// The `Queue` type corresponds to the same data structure.
+library;
+
+use std::vec::Vec;
+
+pub struct Queue<T> {
+    vec: Vec<T>,
+}
+
+impl<T> Queue<T> {
+    /// Create a new `Queue`
+    pub fn new() -> Self {
+        Self {
+            vec: Vec::new(),
+        }
+    }
+
+    /// Checks a `Queue` for emptiness
+    pub fn is_empty(self) -> bool {
+        self.vec.is_empty()
+    }
+
+    /// Gets the number of elements in the `Queue`
+    pub fn len(self) -> u64 {
+        self.vec.len()
+    }
+
+    /// Enqueues a `Queue`
+    pub fn enqueue(ref mut self, item: T) {
```

Comment:
I would add a section for the argument which explains what the `item` is.

Reply:
Added such section